histogram.go

```diff
@@ -0,0 +1,259 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package tally
+
+import (
+	"fmt"
+	"math"
+	"sort"
+	"time"
+)
+
+// DefaultBuckets can be passed to specify to default buckets.
+var DefaultBuckets Buckets
+
+// ValueBuckets is a set of float64 values that implements Buckets.
+type ValueBuckets []float64
+
+// Implements sort.Interface
+func (v ValueBuckets) Len() int {
+	return len(v)
+}
+
+// Implements sort.Interface
+func (v ValueBuckets) Swap(i, j int) {
+	v[i], v[j] = v[j], v[i]
+}
+
+// Implements sort.Interface
+func (v ValueBuckets) Less(i, j int) bool {
+	return v[i] < v[j]
+}
+
+func (v ValueBuckets) String() string {
+	values := make([]string, len(v))
+	for i := range values {
+		values[i] = fmt.Sprintf("%f", v[i])
+	}
+	return fmt.Sprint(values)
+}
+
+// AsValues implements Buckets.
+func (v ValueBuckets) AsValues() []float64 {
+	return []float64(v)
+}
+
+// AsDurations implements Buckets and returns time.Duration
+// representations of the float64 values divided by time.Second.
+func (v ValueBuckets) AsDurations() []time.Duration {
+	values := make([]time.Duration, len(v))
```

Comment:
This is true however none of our other interfaces return an error on a counter inc or gauge update or timer record.  I can perhaps make it a no-op if you `RecordValue(...)` on a histogram that has duration buckets and vice-versa if you feel strongly?